components/kserve/kserve_config_handler.go

```diff
@@ -0,0 +1,163 @@
+package kserve
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/hashicorp/go-multierror"
+	operatorv1 "github.com/openshift/api/operator/v1"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	dsciv1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/dscinitialization/v1"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/deploy"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/feature"
+)
+
+const (
+	KserveConfigMapName string = "inferenceservice-config"
+)
+
+func (k *Kserve) setupKserveConfigAndDependencies(ctx context.Context, cli client.Client, dscispec *dsciv1.DSCInitializationSpec) error {
+	// as long as Kserve.Serving is not 'Removed', we will setup the dependencies
+
+	switch k.Serving.ManagementState {
+	case operatorv1.Managed:
+		// check on dependent operators if all installed in cluster
+		dependOpsErrors := checkDependentOperators(cli).ErrorOrNil()
+		if dependOpsErrors != nil {
+			return dependOpsErrors
+		}
+
+		if err := k.configureServerless(dscispec); err != nil {
+			return err
+		}
```

Comment:
I don't think it is right to have the call to `configureServerless` inside the `switch` block, given the original code does not have any conditions around `Serving.ManagementState` for invoking this func.

The unwanted side-effect I can see is that flipping `Serving.ManagementState` from `Managed` to `Removed` won't lead to the removal of Knative.

Reply:
moved `configureServerless` out of this function and directly called it in the Kserve reconcile loop before Kserve is deployed.